### PR TITLE
Add test coverage of util module

### PR DESCRIPTION
**Pull request by eslavich**
This PR adds test coverage of util.py.

### CODE REVIEW

File: tests/test_util.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    assert result.dtype == np.float32
+    assert_array_equal(result, np.array([3.14159], dtype=np.float32))
+
+
+def test_gentle_asarray_invalid_conversion():
+    with pytest.raises(ValueError):
+        util.gentle_asarray(object(), dtype=np.float32)
+
+
+def test_get_short_doc():
```
Comment (jdavies-st) [inline tests/test_util.py:94]: Is there a reason to have `get_short_doc` is not over in the `fits_support.py` module?  It's only used for populating the FITS keyword comment when writing out files to FITS format I believe.

File: tests/test_util.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        "description": "Some schema description.",
+    }) == "Some schema description."
+    assert util.get_short_doc({
+        "description": "Some schema description.\nWhoops, another line.",
+    }) == "Some schema description."
+
+
+def test_ensure_ascii():
+    for inp in [b"ABCDEFG", "ABCDEFG"]:
+        util.ensure_ascii(inp) == "ABCDEFG"
```
Comment (jdavies-st) [inline tests/test_util.py:114]: Is there a reason to have the `ensure_ascii` function anymore now that we are python3-only?  I.e., we probably shouldn't be passing around bytes as text, and if we are, it's probably because of python2.  But if it really requires `ascii`, then we should be making sure it actually is ascii, and it seems like this function only decodes bytes to ascii.  About unicode?

**Review by jdavies-st — COMMENTED** [review]
Looks good.  Some 1-line comments in the tests would be nice to guide the reader what is being tested.  The names of the tests merely hint.  =)

Some other questions about perhaps outdated code in `ensure_ascii`.

File: tests/test_util.py (hunk 1 above)
Comment (eslavich) [reply tests/test_util.py:94]: It does appear to only be used in fits_support.py, so I can quarantine it over there.

File: tests/test_util.py (hunk 2 above)
Comment (eslavich) [reply tests/test_util.py:114]: That is odd, isn't it.  The code in fits_support.py is using this on values extracted from the schema, which will always be strings anyway.  The FITS standard restricts the characters that may be used in a header:

image

So, maybe this method should actually be ensuring characters that fall within the allowed range?

File: tests/test_util.py [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    assert result.dtype == np.int8
+    assert_array_equal(result, inp)
+
+    dtype = np.dtype([("col1", np.int16), ("col2", np.float32)])
+    result = util.gentle_asarray(inp, dtype=dtype)
+    assert result.dtype == dtype
+    assert result["col1"].dtype == np.int16
+    assert_array_equal(result["col1"], np.array([1, 2, 3, 4], dtype=np.int16))
+    assert result["col2"].dtype == np.float32
+    assert_array_equal(result["col1"], np.array([1, 2, 3, 4], dtype=np.float32))
```
[the remark this answers is not in the document]
Comment (eslavich) [reply tests/test_util.py:29]: Yes, broadcasting out to a recarray with multiple columns from a regular array.  This hits a corner case in the `gentle_asarray` code that wasn't getting tested otherwise.  I'll add notes to this and the rest of the tests like you suggested.

File: tests/test_util.py (hunk 2 above)
Comment (eslavich) [reply tests/test_util.py:162]: I'm going to punt on this by also moving it to fits_support.py, which I'm planning to add tests for in a future PR.

**Review by jdavies-st — APPROVED** [review]
🚀
